tests/unit/App.spec.js

```diff
@@ -89,4 +89,55 @@ Given("an App", () => {
       expect(app.vm.url).to.equal(expectedUrl);
     });
   });
+
+  When("a response is returned", () => {
+    beforeEach(async () => {
+      app.setData({
+        wasResponseAnError: false,
+        statusText: ""
+      });
+
+      app.setMethods({
+        request: app.vm.handleResponse
+      });
+    });
+
+    And("the response was a 4xx", () => {
+      const response = {
+        status: 404,
+        statusText: "Not Found"
+      };
+
+      beforeEach(async () => {
+        app.vm.request(response);
```

Comment:
I don't understand how you can really call this testing your request function. You're just testing your handleResponse function. I thought you were going to mock fetch??

Reply:
My understanding was that I don't need to use any mocking frameworks at this time. I thought that I should just set the `request` method to be the method I would use as my handler (`handleResponse`), and that I could test that `handleResponse` does what it is supposed to.

If this isn't correct, could you please clarify what you mean by "mocking fetch"?

Reply:
I don't think mocking fetch is necessary. This is not testing the request function, it's testing that the error handling. It is, however, testing the implementation rather than the behavior. If you chose the represent an error state in App differently, your tests would break.

Reply:
So I tried using selectors to access the properties of the `Code` child component (to test that the classList contains "border-danger", and the text() contains "Not Found"). However, I wasn't able to find the right combination of selectors (I'm not sure if you can even `find` an element on a sub-component).